renderdoc/driver/ihv/amd/amd_counters.cpp

```diff
@@ -227,140 +414,374 @@ CounterDescription AMDCounters::InternalGetCounterDescription(uint32_t internalI
 
 void AMDCounters::EnableCounter(GPUCounter counter)
 {
-  const uint32_t internalIndex = m_Counters[GPUCounterToCounterIndex(counter)].internalIndex;
+  const uint32_t internalIndex = m_PublicToInternalCounter[counter];
 
-  m_pGPUPerfAPI->GPA_EnableCounter(internalIndex);
+  auto status = m_pGPUPerfAPI->GPA_EnableCounter(internalIndex);
+  if (AMD_FAILED(status))
+  {
+    GPA_LoggingCallback(GPA_LOGGING_ERROR, FormatErrMessage("Enable counter.", status).c_str());
+  }
 }
 
 void AMDCounters::EnableAllCounters()
 {
-  m_pGPUPerfAPI->GPA_EnableAllCounters();
+  auto status = m_pGPUPerfAPI->GPA_EnableAllCounters();
+  if (AMD_FAILED(status))
+  {
+    GPA_LoggingCallback(GPA_LOGGING_ERROR, FormatErrMessage("Enable all counters.", status).c_str());
+  }
 }
 
 void AMDCounters::DisableAllCounters()
 {
-  m_pGPUPerfAPI->GPA_DisableAllCounters();
+  auto status = m_pGPUPerfAPI->GPA_DisableAllCounters();
+  if (AMD_FAILED(status))
+  {
+    GPA_LoggingCallback(GPA_LOGGING_ERROR, FormatErrMessage("Disable all counters.", status).c_str());
+  }
 }
 
 uint32_t AMDCounters::GetPassCount()
 {
-  gpa_uint32 numRequiredPasses;
-  m_pGPUPerfAPI->GPA_GetPassCount(&numRequiredPasses);
+  gpa_uint32 numRequiredPasses = 0;
+  auto status = m_pGPUPerfAPI->GPA_GetPassCount(&numRequiredPasses);
+  if (AMD_FAILED(status))
+  {
+    GPA_LoggingCallback(GPA_LOGGING_ERROR, FormatErrMessage("Get pass count.", status).c_str());
+  }
 
   return (uint32_t)numRequiredPasses;
 }
 
 uint32_t AMDCounters::BeginSession()
 {
-  gpa_uint32 sessionID;
+  gpa_uint32 sessionID = 0;
 
-  m_pGPUPerfAPI->GPA_BeginSession(&sessionID);
+  auto status = m_pGPUPerfAPI->GPA_BeginSession(&sessionID);
+  if (AMD_FAILED(status))
+  {
+    GPA_LoggingCallback(GPA_LOGGING_ERROR, FormatErrMessage("Begin session.", status).c_str());
+  }
 
   return (uint32_t)sessionID;
 }
 
 void AMDCounters::EndSesssion()
 {
-  m_pGPUPerfAPI->GPA_EndSession();
+  auto status = m_pGPUPerfAPI->GPA_EndSession();
+  if (AMD_FAILED(status))
+  {
+    GPA_LoggingCallback(GPA_LOGGING_ERROR, FormatErrMessage("End session.", status).c_str());
+  }
+}
+
+std::vector<CounterResult> AMDCounters::GetCounterData(
+  uint32_t sessionID,
+  uint32_t maxSampleIndex,
+  const std::vector<uint32_t> &eventIDs,
+  const std::vector<GPUCounter> &counters
+)
+{
+  std::vector<CounterResult> ret;
+
+  bool isReady = false;
+
+  const uint32_t timeout = 10000; // ms
+
+  auto startTime = std::chrono::high_resolution_clock::now();
```

Comment:
`PerformanceTimer` gives an abstracted timer, which should be used here instead of STL constructs.

Reply:
Fixed